### PR TITLE
feat(storage): reduce compactor memory

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
Rest LGTM

File: src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs [hunk 1]
```diff
@@ -205,6 +206,12 @@ impl NonOverlapSubLevelPicker {
                 break;
             }
 
+            if ret.total_file_size >= self.max_compaction_bytes
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:209]: A mistake. I will remove this code.

File: src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs [hunk 2]
```diff
@@ -289,6 +296,12 @@ impl NonOverlapSubLevelPicker {
                 && ret.total_file_size + (add_files_size + current_level_size)
                     >= self.max_compaction_bytes
                 || ret.total_file_count + add_files_count >= self.max_file_count as usize
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:298]: No...
ret
                    .sstable_infos
                    .iter()
                    .filter(|ssts| !ssts.is_empty())
                    .count()
                    > MAX_LEVEL_COUNT) means that there will be at least one sub-level selected

File: src/common/src/system_param/mod.rs [hunk 3]
```diff
@@ -46,7 +46,8 @@ macro_rules! for_all_undeprecated_params {
         $macro! {
             { barrier_interval_ms, u32, Some(1000_u32), true },
             { checkpoint_frequency, u64, Some(1_u64), true },
-            { sstable_size_mb, u32, Some(256_u32), false },
+            { sstable_size_mb, u32, Some(64_u32), false },
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply src/common/src/system_param/mod.rs:49]: To release memory of flush as soon as possible

**Review by Li0k — APPROVED** [review]

**Pull request merged**